src/main/cpp/subsystems/ExtensionSubsystem.cpp

```diff
@@ -1,23 +1,68 @@
 #include "subsystems/ExtensionSubsystem.h"
 
-ExtensionSubsystem::ExtensionSubsystem() {
-    // Implementation of subsystem constructor goes here.
-
-    m_extensionMotor.EnableSoftLimit(
-        rev::CANSparkMax::SoftLimitDirection::kForward, true);
+ExtensionSubsystem::ExtensionSubsystem()
+    : BaseSingleAxisSubsystem(m_config, m_extensionMotor, m_encoder, &min,
+                              nullptr, "EXTEND", "\033[96;40;4m") {
+    m_extensionMotor.SetInverted(true);
+    _config = m_config;
+    _controller = m_config.pid;
+    _controller.SetTolerance(.5, 1);
+    _config.distancePerRevolution = ArmConstants::kInPerRotation;
+    m_encoder.SetPositionConversionFactor(1);
+}
 
-    // Default unit is number of rotations
-    m_extensionMotor.SetSoftLimit(
-        rev::CANSparkMax::SoftLimitDirection::kForward,
-        ArmConstants::kExtenderSoftLimit);
+void ExtensionSubsystem::ResetEncoder() {
+    if (_log)
+        Logging::logToStdOut(_prefix, "RESET POSITION", Logging::Level::INFO);
+    m_encoder.SetPosition(0);
 }
 
-void ExtensionSubsystem::Periodic() {}
+double ExtensionSubsystem::GetCurrentPosition() {
+    auto position = m_encoder.GetPosition() * _config.distancePerRevolution;
 
-void ExtensionSubsystem::SimulationPeriodic() {}
+    Logging::logToSmartDashboard("ExtensionPosition",
+                                 std::to_string(position) + " in",
+                                 Logging::Level::INFO);
 
-void ExtensionSubsystem::PercentOutput(double output) {
-    m_extensionMotor.Set(output);
+    if (_log)
+        Logging::logToStdOut(_prefix,
+                             std::to_string(position) + "/" +
+                                 std::to_string(_config.maxDistance) + " in",
+                             Logging::Level::INFO);
+
+    return position;
 }
 
-void ExtensionSubsystem::Extend(double speed) { m_extensionMotor.Set(speed); }
\ No newline at end of file
+void ExtensionSubsystem::UpdateMovement() {
```

Comment:
Try to consolidate all of these `UpdateMovement()` overrides back into the base class and use the config instead.